app/services/workflow_link_service.rb

```diff
@@ -1,25 +1,33 @@
+require 'remote_tagging_service'
 class WorkflowLinkService
   attr_accessor :workflow_id
 
-  TAG_PREFIX = "/approval/workflows/".freeze
+  TAG_NAMESPACE = 'approval'.freeze
+  TAG_NAME      = 'workflows'.freeze
 
   def initialize(workflow_id)
     self.workflow_id = workflow_id
   end
 
   def link(tag_attrs)
     TagLink.find_or_create_by!(tag_link(tag_attrs))
+    RemoteTaggingService.new(tag_attrs).process('add', approval_tag)
     nil
   end
 
   private
 
   def tag_link(tag_attrs)
-    tag_attrs.except(:object_id).merge(:workflow_id => workflow_id, :tag_name => tag_name)
+    tag_attrs.except(:object_id).merge(:workflow_id => workflow_id, :tag_name => fq_tag_name)
   end
 
-  # TODO: create tag name based on workflow id
-  def tag_name
-    "#{TAG_PREFIX}#{workflow_id}"
+  def fq_tag_name
+    "/#{TAG_NAMESPACE}/#{TAG_NAME}=#{workflow_id}"
+  end
+
+  def approval_tag
+    { :name      => TAG_NAME,
+      :value     => workflow_id.to_s,
```

Comment:
Will all remote services allow the same namespace and name to have multiple values?

Reply:
Yes. If the name and namespace remains the same but the value changes it will result in different tags being created. If you add the same tag multiple times with the same name, namespace and value it doesn't add the same thing multiple times.